lib/Backends/CPU/libjit/libjit.cpp

```diff
@@ -1089,6 +1089,31 @@ void libjit_sparse_lengths_weighted_sum_f(float *dest, float *data,
   }
 }
 
+void libjit_sparse_lengths_weighted_sum_grad_f(float *destGrad, float *dataGrad,
+                                               float *weights, size_t *indices,
+                                               int32_t *lengths,
+                                               size_t segments, size_t lineSize,
+                                               size_t numLines) {
+  // The data gradients not touched by this operation should
+  // be 0, so set the entire buffer to 0 to start with.
+  memset(dataGrad, 0, numLines * lineSize * sizeof(float));
```

Comment:
nit: Since `numLines` is only used here, perhaps calculate the size for `memset()` at compile time in LLVMIRGen and pass it instead of `numLines`.